File: 백준 문제풀이/2567.py
'''
배열 만들어 놓고 칠하기
현재 자신이 0이고 상하좌우 중 한곳이 1이라면 테두리
테두리 카운트 해서 출력
'''
dx = [0,0,-1,1]
dy = [-1,1,0,0]
result = 0
board = [[0]*101 for _ in range(101)]

for _ in range(int(input())):
    s,e = map(int,input().split())
    # 채우기 시작
    for i in range(s,s+10):
        for j in range(e,e+10):
            board[i][j] = 1 # 채우기

# 몇개인지 카운팅
for i in range(101):
    for j in range(101):
        if board[i][j] == 1: # 0을 발견했고,상하좌우 중에 1이존재한다면
            for k in range(4):
                ni = i + dx[k]
                nj = j + dy[k]
                if 0<=ni<101 and 0<=nj<101:
                    if board[ni][nj] == 0:
                        result+=1
                
                
print(result)

File: 백준 문제풀이/2567.cpp
#include <bits/stdc++.h>
using namespace std;
int board[101][101];
int dx[4] = {0,0,-1,1};
int dy[4] = {-1,1,0,0};
int result = 0;
/**
 * 전체 도화지는 100*100 크기
 * 색종이는 10*10
 * 색종이가 도화지 밖으로 나가는 경우는 없다
 * 검은색 영역 구하기
 * 
*/

// 검은색 영역으로 칠하기
void draw(int s,int e){
    for(int i=s;i<s+10;i++){
        for(int j=e;j<e+10;j++){
            board[i][j] = 1;
        }
    }
    
}

// 자신이 0이고, 상,하,좌,우 중 하나에 1이 존재한다면 테두리
void count(){
    for(int i=0;i<101;i++){
        for(int j=0;j<101;j++){
            if(board[i][j]!=1)
                continue;
            bool isOne = false;
            for(int k=0;k<4;k++){
                int ni = i + dx[k];
                int nj = j + dy[k];
                if(0<=ni&&ni<=100&&0<=nj&&nj<=100){
                    // 1이 존재하는지 확인
                    if(board[ni][nj] == 1){
                        isOne = true;
                        break;
                    }
                }
            }
            if(isOne)
                result++;
        }
    }
}

int main(){
    int t,s,e; cin>>t;
    while(t--){
        cin>>s>>e;
        draw(s,e);
    }

    for(int i =0;i<101;i++){
        for(int j=0;j<101;j++){
            cout<<board[i][j];
        }
        cout<<endl;
    }
    count();
    cout<<result;
    
}